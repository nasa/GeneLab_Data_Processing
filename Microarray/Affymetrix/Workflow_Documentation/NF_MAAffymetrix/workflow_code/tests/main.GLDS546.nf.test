nextflow_pipeline {

    name "Test Workflow main.nf"
    script "main.nf"
    tag "core"

    test("GLDS-546: Should run without failures") {

        when {
            params {
                // define parameters here. Example: 
                osdAccession = "OSD-546"
                gldsAccession = "GLDS-546"
                // biomart_attribute = "AFFY HuGene 1 0 st v1" IN ISA
                limit_biomart_query = 300
            }
        }

        then {
            assert workflow.success
            assert snapshot(
                            path("${ params.outputDir }/Metadata").list(),
                            path("${ params.outputDir }/00-RawData").list(),
                            path("${ params.outputDir }/01-oligo_NormExp").list(),
                            path("${ params.outputDir }/02-limma_DGE").list(),
                            ).match()
        }

    }

}
