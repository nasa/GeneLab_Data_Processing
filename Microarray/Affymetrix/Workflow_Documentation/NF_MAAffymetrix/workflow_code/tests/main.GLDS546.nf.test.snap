{
    "GLDS-546: Should run without failures": {
        "content": [
            [
                "OSD-546_metadata_OSD-546-ISA.zip:md5,794dfe5d7af2583196545d618b517d21",
                "OSD-546_microarray_v0_runsheet.csv:md5,06eda1b290b0a480e5060ca53fc4fcda"
            ],
            [
                "raw_intensities_probe.csv:md5,b0a837ca6bc4a9ca0b59fbf86e73cd06"
            ],
            [
                "normalized_expression_probeset.csv:md5,550830946e555453cd233d87b54798eb",
                "normalized_intensities_probe.csv:md5,acc7f88e9cbac432711ca32813e1a31f"
            ],
            [
                "SampleTable.csv:md5,a979461af3c12ea30d3676e1592b9806",
                "contrasts.csv:md5,8e7e04b2c266b84f30acaeb21d15debb",
                "differential_expression.csv:md5,9bafc26e410e5701378729fa3d1a17ae",
                "visualization_PCA_table.csv:md5,f70064380839b0703983964412b3bc38"
            ]
        ],
        "timestamp": "2023-05-17T19:19:32+0000"
    }
}